Sources/SocksCore/TCPSocket.swift

```diff
@@ -104,6 +104,10 @@ public class TCPInternetSocket: InternetSocket, TCPSocket, TCPReadableSocket, TC
 
         try setReuseAddress(true)
     }
+    
+    deinit {
+        try? self.close()
```

Comment:
Do we run any risk of exhausting open file handles if things are not closed properly and we are not warned? Perhaps we should log an error here.